File: src/main/java/al/uys/project_demo/Commons/MessageResponse.java
package al.uys.project_demo.Commons;

import al.uys.project_demo.Commons.enums.MessageResponseType;
import lombok.Getter;
import lombok.RequiredArgsConstructor;

@Getter
@RequiredArgsConstructor
public class MessageResponse extends Throwable {
  private final String message;
  private final MessageResponseType messageResponseType;

  public boolean hasError() {
    return messageResponseType.equals(MessageResponseType.ERROR);
  }
}

File: mobile/lib/helpers/MessageResponse.dart

class MessageResponse {
  final message;
  final messageResponseType;

  MessageResponse({required this.message, required this.messageResponseType});

  factory MessageResponse.fromJson(Map<String, String> json) {
    return new MessageResponse(
      message: json['message'],
      messageResponseType: json['messageResponseType'],
    );
  }
}
